src/wallet/wallet2.cpp

```diff
@@ -3138,6 +3137,51 @@ bool wallet2::get_rct_distribution(uint64_t &start_height, std::vector<uint64_t>
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool wallet2::get_output_blacklist(std::vector<uint64_t> &blacklist)
+{
+  uint32_t rpc_version;
+  boost::optional<std::string> result = m_node_rpc_proxy.get_rpc_version(rpc_version);
+  // no error
+  if (!!result)
```

Comment:
Is there any reason to prefer this over `if(result)`?

Reply:
Copy pasta. I just made it ```if (result)``` now and updated some of the error messages there to be relevant for output blacklist.